File: Cuarto/Laboratorio/Talleres/RobocodeLabo/src/Robotito2/CombatStrategy.java
package Robotito2;

import robocode.*;

public interface CombatStrategy {
     double bulletHalfPower = 1.5;
     double bulletMaxPower = 3;
     int terminatorThreshold = 5;
     int leftWallAngle = 270;

    public abstract void init(JuniorRobot robot);

    public abstract void onTick(JuniorRobot robot);

    public abstract void onScannedRobot(JuniorRobot robot);

    public abstract void onHitWall(JuniorRobot robot);

    public abstract void onHitRobot(JuniorRobot robot);

    public abstract void onHitByBullet(JuniorRobot robot);
}

File: Cuarto/Laboratorio/Talleres/RobocodeLabo/src/RobotitoKotlin/CombatStrategy.kt
package RobotitoKotlin
import robocode.*;

abstract class CombatStrategy(protected val robot: JuniorRobot) {


    protected val bulletHalfPower = 1.5
    protected val bulletMaxPower = 3.0

    abstract fun onTick()
    abstract fun onScannedRobot()
    abstract fun onHitWall()
    abstract fun onHitRobot()
    abstract fun onHitByBullet()

    open fun readyToAttack(): Boolean {
        return true
    }
}
